autopxd/__init__.py

```diff
@@ -34,11 +36,92 @@ def ensure_binary(s, encoding="utf-8", errors="strict"):
     raise TypeError(f"not expecting type '{type(s)}'")
 
 
+def _find_cl():
+    """Use vswhere.exe to locate the Microsoft C compiler."""
+    host_platform = {
+        "X86": "X86",
+        "AMD64": "X64",
+        "ARM64": "ARM64",
+    }.get(platform.machine(), "x86")
```

Comment:
Should this default to uppercase? Or should they all be lowercase?

```suggestion
    }.get(platform.machine(), "X86")
```